components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/saml/SAML2BearerGrantHandler.java

```diff
@@ -887,6 +887,92 @@ private void validateNameId(OAuthTokenReqMessageContext tokReqMsgCtx, Assertion
         }
     }
 
+    /**
+     * Get the user id of the identity provider.
+     *
+     * @param tokReqMsgCtx
+     * @param identityProvider
+     * @param assertion
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    protected String getUserId(OAuthTokenReqMessageContext tokReqMsgCtx, IdentityProvider identityProvider,
+                             Assertion assertion) throws IdentityOAuth2Exception {
+
+        // Check whether the user id from claims is enabled for the SAML bearer grant.
+        if (OAuthServerConfiguration.getInstance().getSaml2UserIdFromClaims()) {
```

Comment:
why are we checking a server configuration? shouldn't this be configured in the IDP configuration? That's what I understood from the issue description.

Reply:
This configuration had been kept in order to keep backward compatible. If some one want to the previous behaviour (If some one want to use the old behaviour that Use NameID always for the saml bearer grant without depending on the IDP configuration then they could disable the new feature suing the property configured in identity.xml)